### PR TITLE
Refactor/add tooltip to playground

### LINKED ISSUE #54619: Add hover tooltip on download button in Playground
### Describe the problem that you experienced

As we know the cool angular.dev playground has a feature to download the provided/written source code. 
I find that at the first time of seeing/using this button it was pretty unclear to me what it does. First I thought it will just download the playground file I have currently open.

### Enter the URL of the topic with the problem

https://angular.dev/playground

### Describe what you were looking for in the documentation

_No response_

### Describe the actions that led you to experience the problem

_No response_

### Describe what you want to experience that would fix the problem

I suggest a tooltip below the button, that displays when hovering over the button, which says `Download the current application source code` or similar.

https://material.angular.io/components/tooltip/overview

### Add a screenshot if that helps illustrate the problem

image

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

_No response_

### Provide any additional information here in as much as detail as you can

I'd be happy to implement this UX/Accessibility improvement. 😄

### PR DESCRIPTION
**Pull request by ColinJolivet**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.dev application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Fixes #54619 

## What is the new behavior?

When hovering the download button in the Angular playground, it now display a tooltip.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by MarkTechson** [comment]
@ColinJolivet looks like there are a couple of failing steps, one lint check and a failing test. Can you please take another look?

**Comment by ColinJolivet** [comment]
The lint check error is related to the lack of scope in the header of my commit message, I thought it was optional as it says in the contributing guideline. I'll correct this by adding a scope in my next commit.

But for the test failure, I'm not sure if it's related to what I did, and I can't reproduce the error in my environment, so I'm not sure what I can do to solve this problem. I'll look into it a bit further.

**Comment by JeanMeche** [comment]
The test failure a is flaky test. Re-running it should fix it !

**Comment by ColinJolivet** [comment]
Thanks !

It did indeed solve the problem.

As for the lint problem, since I've already pushed the commit, do I have a way to modify it this way, or do I have to make a new one without any changes?

File: adev/src/app/editor/code-editor/code-editor.component.spec.ts [hunk 1]
```diff
@@ -51,7 +53,7 @@ describe('CodeEditor', () => {
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [CodeEditor, NoopAnimationsModule],
+      imports: [CodeEditor, NoopAnimationsModule, MatTooltipModule],
```
Comment (JeanMeche) [inline adev/src/app/editor/code-editor/code-editor.component.spec.ts:56]: Ideally we'd prefer having the standalone directive instead of the whole module.

**Comment by JeanMeche** [comment]
The linting job is triggered manually on our side. 
It will fail because of the merge commit. We'd like this change to be a single commit (this will require a drop & a squash).

File: adev/src/app/editor/code-editor/code-editor.component.spec.ts (hunk 1 above)
Comment (ColinJolivet) [reply adev/src/app/editor/code-editor/code-editor.component.spec.ts]: I change not to use MatTooltipModule but instead just MatTooltip

I'm not sure if it was what you wanted, if not enlighten me

File: adev/src/app/editor/code-editor/code-editor.component.ts [hunk 2]
```diff
@@ -48,7 +49,15 @@ const ANGULAR_DEV = 'https://angular.dev';
   templateUrl: './code-editor.component.html',
   styleUrls: ['./code-editor.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
-  imports: [MatTabsModule, IconComponent, ClickOutside, CdkMenu, CdkMenuItem, CdkMenuTrigger],
+  imports: [
+    MatTabsModule,
+    MatTooltip,
```
[the remark this answers is not in the document]
Comment (ColinJolivet) [reply adev/src/app/editor/code-editor/code-editor.component.ts:54]: Maybe I didn't understand what you wanted from me, but I feel like I'm not using the module.

If I'm wrong, can you please explain me a bit more ?

**Comment by ColinJolivet** [comment]
The linting problem is resolve, i added the scope. Also, i squashed all the commits in one.

**Comment by JeanMeche** [comment]
Could you make it a `refactor` commit. We keep `feat` for framework features. Thank you.

**Comment by ColinJolivet** [comment]
Done !

It is now a refactor commit

**Comment by JeanMeche** [comment]
There are some styling issues (there is not backgrond, the font size seems a bit large etc). We're probably missing an `@include mat-tooltip-theme($theme)` in `adev/shared-docs/styles/global-styles.scss`

Screenshot 2024-10-04 at 14 54 49

**Comment by ColinJolivet** [comment]
Indeed, the ```@include mat-tooltip-theme($theme)``` was missing, i added it.

**Review by JeanMeche — APPROVED** [review]
Thanks for this !

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 5e0029237cebc3a32a25772933d0148d49dd7100.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**